### PR TITLE
Simplify scheduler codes

### PR DESCRIPTION
**Pull request by YangKeao**
- remove persistent config
- remove label and related score
- remove `leveldb_kv`
- remove extra filters:
    - snapshotcount
    - pending peer count
    - storage
- simplify balance region score calculation

And fix related tests.

### CODE REVIEW

File: scheduler/server/schedule/range_cluster.go [hunk 1]
```diff
@@ -27,21 +27,6 @@ type RangeCluster struct {
 	tolerantSizeRatio float64
 }
 
-const scanLimit = 128
-
-// GenRangeCluster gets a range cluster by specifying start key and end key.
-// The cluster can only know the regions within [startKey, endKey].
-func GenRangeCluster(cluster opt.Cluster, startKey, endKey []byte) *RangeCluster {
```
Comment (Connor1996) [inline scheduler/server/schedule/range_cluster.go]: remove `RangeCluster` totally?

File: scheduler/server/schedulers/balance_leader.go [hunk 2]
```diff
@@ -1,193 +0,0 @@
...
-
-	schedule.RegisterScheduler("balance-leader", func(opController *schedule.OperatorController, storage *core.Storage, mapper schedule.ConfigDecoder) (schedule.Scheduler, error) {
-		return newBalanceLeaderScheduler(opController), nil
-	})
-}
-
-// balanceLeaderRetryLimit is the limit to retry schedule for selected source store and target store.
-const balanceLeaderRetryLimit = 10
-
-type balanceLeaderScheduler struct {
```
Comment (Connor1996) [inline scheduler/server/schedulers/balance_leader.go]: please keep `balanceLeaderScheduler` though it's not included in the course outline.

File: scheduler/server/server.go [hunk 3]
```diff
@@ -556,76 +536,17 @@ func (s *Server) SetReplicationConfig(cfg config.ReplicationConfig) error {
 func (s *Server) SetPDServerConfig(cfg config.PDServerConfig) error {
```
Comment (Connor1996) [inline scheduler/server/server.go]: remove these API too, we don't need to change configs dynamically.

File: scheduler/server/server.go [hunk 4]
```diff
@@ -556,76 +536,17 @@ func (s *Server) SetReplicationConfig(cfg config.ReplicationConfig) error {
...
-	return nil
-}
-
-// GetLabelProperty returns the whole label property config.
-func (s *Server) GetLabelProperty() config.LabelPropertyConfig {
-	return s.scheduleOpt.LoadLabelPropertyConfig().Clone()
-}
-
 // SetClusterVersion sets the version of cluster.
 func (s *Server) SetClusterVersion(v string) error {
```
Comment (Connor1996) [inline scheduler/server/server.go]: Please clean all things related with cluster version

File: scheduler/server/cluster_test.go [hunk 5]
```diff
@@ -1143,34 +1068,6 @@ func (s *testClusterInfoSuite) TestRegionHeartbeat(c *C) {
 	}
 }
 
-func heartbeatRegions(c *C, cluster *RaftCluster, regions []*core.RegionInfo) {
```
Comment (Connor1996) [inline scheduler/server/cluster_test.go]: why remove this?

File: scheduler/server/cluster_worker_test.go [hunk 6]
```diff
@@ -90,42 +88,3 @@ func (s *testClusterWorkerSuite) TestValidRequestRegion(c *C) {
 	c.Assert(cluster.validRequestRegion(r5), IsNil)
 	cluster.stop()
 }
-
-func (s *testClusterWorkerSuite) TestAskSplit(c *C) {
```
Comment (Connor1996) [inline scheduler/server/cluster_worker_test.go]: please keep it

Comment (YangKeao) [reply scheduler/server/cluster_worker_test.go]: Oops! forget to restore it.

File: scheduler/server/config/config.go [hunk 7]
```diff
@@ -419,6 +419,8 @@ func (c *Config) configFromFile(path string) (*toml.MetaData, error) {
 
 // ScheduleConfig is the schedule configuration.
 type ScheduleConfig struct {
+	// SplitMergeInterval is the minimum interval time to permit merge after split.
+	SplitMergeInterval typeutil.Duration `toml:"split-merge-interval,omitempty" json:"split-merge-interval"`
```
Comment (Connor1996) [inline scheduler/server/config/config.go]: this can be deleted

**Pull request merged**